src/Components/Server/src/Circuits/CircuitHost.cs

```diff
@@ -95,15 +95,23 @@ public static void SetCurrentCircuitHost(CircuitHost circuitHost)
 
         public IDispatcher Dispatcher { get; }
 
-        public Task<IEnumerable<string>> PrerenderComponentAsync(Type componentType, ParameterCollection parameters)
+        public Task<ComponentRenderedText> PrerenderComponentAsync(Type componentType, ParameterCollection parameters)
         {
             return Dispatcher.InvokeAsync(async () =>
             {
                 var result = await Renderer.RenderComponentAsync(componentType, parameters);
+
+                await OnCircuitOpenedAsync(CancellationToken.None);
```

Comment:
What's the rationale for this change? I've seen this code move around now in multiple PRs - leave a comment?